cucumber/authenticators_jwt/features/support/jwt_jwks_helper.rb

```diff
@@ -91,10 +128,60 @@ def issue_jwt_token_not_memoized_key(token_body, algorithm = Algorithms::RS256)
     )
   end
 
+  def issue_jwt_ecdsa_token(token_body, algorithm = Algorithms::ES256)
+    @jwt_token = JWT.encode(
+      token_body,
+      ecdsa_private_key,
+      algorithm,
+      {
+        kid: jwk_ecdsa.kid
+      }
+    )
+  end
+
+  def issue_jwt_hmac_token(token_body, algorithm = Algorithms::HS256)
+    @jwt_token = JWT.encode(
+      token_body,
+      HMAC_SECRET,
+      algorithm
+    )
+  end
+
+  def issue_rsa_jwt_token_with_alg_header(token_body, alg_header, algorithm = Algorithms::RS256)
+    encoded_token = JWT.encode(
+      token_body,
+      rsa_keys[@default_file_name],
+      algorithm
+    )
+    token_parts = encoded_token.split(".")
+    headers = "{\"typ\":\"JWT\",\"alg\":\"#{alg_header}\",\"kid\":\"#{jwk_set[@default_file_name].kid}\"}"
+    token_parts[0] = Base64.urlsafe_encode64(headers).strip
+    @jwt_token = token_parts.join(".")
```

Comment:
You manipulate token after signature. Means it's invalid by definition.
Wrong algorithm has to be signed.

Reply:
But this is what i want to test here

Reply:
Lets discuss it

Reply:
I'm not sure that this's the purpose of the test